src/main/telemetry/ibus.c

```diff
@@ -0,0 +1,387 @@
+}
+
+static void transmitIbusPacket(uint8_t ibusPacket[static IBUS_MIN_LEN], size_t packetLength) {
+    uint16_t checksum = calculateChecksum(ibusPacket, packetLength);
+    ibusPacket[packetLength - IBUS_CHECKSUM_SIZE] = (checksum & 0xFF);
+    ibusPacket[packetLength - IBUS_CHECKSUM_SIZE + 1] = (checksum >> 8);
+    for (size_t i = 0; i < packetLength; i++) {
+        serialWrite(ibusSerialPort, ibusPacket[i]);
+    }
+}
+
+static void sendIbusCommand(ibusAddress_t address) {
+    uint8_t sendBuffer[] = { 0x04, IBUS_COMMAND_DISCOVER_SENSOR | address, 0x00, 0x00 };
+    transmitIbusPacket(sendBuffer, sizeof sendBuffer);
+}
+
+static void sendIbusSensorType(ibusAddress_t address) {
+    uint8_t sendBuffer[] = {0x06,
+                            IBUS_COMMAND_SENSOR_TYPE | address,
+                            SENSOR_ADDRESS_TYPE_LOOKUP[address - ibusBaseAddress],
+                            0x02, 0x0, 0x0 };
+    transmitIbusPacket(sendBuffer, sizeof sendBuffer);
+}
+
+static void sendIbusMeasurement(ibusAddress_t address, uint16_t measurement) {
+    uint8_t sendBuffer[] = { 0x06, IBUS_COMMAND_MEASUREMENT | address, measurement & 0xFF, measurement >> 8, 0x0, 0x0 };
+    transmitIbusPacket(sendBuffer, sizeof sendBuffer);
+}
+
+static bool isCommand(ibusCommand_e expected, uint8_t ibusPacket[static IBUS_MIN_LEN]) {
+    return (ibusPacket[1] & 0xF0) == expected;
+}
+
+static ibusAddress_t getAddress(uint8_t ibusPacket[static IBUS_MIN_LEN]) {
+    return (ibusPacket[1] & 0x0F);
+}
+
+static void dispatchMeasurementRequest(ibusAddress_t address) {
+    switch (SENSOR_ADDRESS_TYPE_LOOKUP[address - ibusBaseAddress]) {
+    case IBUS_SENSOR_TYPE_EXTERNAL_VOLTAGE:
+        {
+            uint16_t value = vbat * 10;
+            if (ibusTelemetryConfig()->report_cell_voltage) {
+                value /= batteryCellCount;
+            }
+            sendIbusMeasurement(address, value);
+        }
+        break;
+
+    case IBUS_SENSOR_TYPE_TEMPERATURE:
+        {
+            #ifdef BARO
+                float temperature = (baroTemperature + 50) / 100.f;
+            #else
+                float temperature = telemTemperature1 / 10.f;
+            #endif
+            sendIbusMeasurement(address, (uint16_t) ((temperature + 40)*10));
+        }
+        break;
+
+    case IBUS_SENSOR_TYPE_RPM:
+        sendIbusMeasurement(address, (uint16_t) rcCommand[THROTTLE]);
+        break;
+    }
+}
+
+static void respondToIbusRequest(uint8_t ibusPacket[static IBUS_RX_BUF_LEN]) {
+    ibusAddress_t returnAddress = getAddress(ibusPacket);
+    
+    //autodetect first received address as our base address
+    if ((INVALID_IBUS_ADDRESS == ibusBaseAddress) && 
+        (INVALID_IBUS_ADDRESS != returnAddress))
+    {
+        ibusBaseAddress = returnAddress;
+    }
+
+    if ((returnAddress >= ibusBaseAddress) &&
+        (ibusAddress_t)(returnAddress - ibusBaseAddress) < sizeof SENSOR_ADDRESS_TYPE_LOOKUP) {
```

Comment:
better use `ARRAYLEN(SENSOR_ADDRESS_TYPE_LOOKUP)`, it will handle larger types corectly